File: spaces-spa/src/app/_models/Post.ts
import { Comment } from "./Comment";
import { LikesForPost } from "./LikesForPost";

export interface Post {
  id: number;
  title: string;
  content: string;
  comments: Comment[];
  dateCreated: Date;
  dateUpdated: Date;
  likedByUsers: LikesForPost[];
  createdBy: string;
}

File: Spaces.DAL/Entities/Post.cs
using System;
using System.Collections.Generic;

namespace Spaces.DAL.Entities;

public class Post
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Content { get; set; }
    public DateTime DateCreated { get; set; }
    public DateTime DateUpdated { get; set; }
    public ICollection<Comment> Comments { get; set; }
    public int UserId { get; set; }
    public AppUser User { get; set; }
    public ICollection<LikeForPost> LikedByUsers { get; set; }
}